library/Icingadb/Web/Control/SearchBar/ObjectSuggestions.php

```diff
@@ -244,70 +250,81 @@ protected function fetchColumnSuggestions($searchTerm)
-        if (! empty($exactVarMatches)) {
-            $this->addHtml(HtmlElement::create(
-                'li',
-                ['class' => static::SUGGESTION_TITLE_CLASS],
-                t('Best Suggestions')
-            ));
+                        $exactVarMatches[] = $varRelation;
 
-            foreach ($exactVarMatches as $relation => $label) {
-                yield $relation => $label;
+                        yield $varRelation => sprintf($label, $name);
+                    }
+                }
             }
         }
 
         // Ordinary columns comes after exact matches,
         // or if there ar no exact matches they come first
-        $colCount = 0;
+        $titleAdded = false;
         foreach (self::collectFilterColumns($model, $query->getResolver()) as $columnName => $columnMeta) {
-            if ($colCount > $colLimit) {
-                break;
-            }
-
             if ($this->matchSuggestion($columnName, $columnMeta, $searchTerm)) {
-                if ($colCount === 0) {
+                if ($titleAdded === false) {
                     $this->addHtml(HtmlElement::create(
                         'li',
                         ['class' => static::SUGGESTION_TITLE_CLASS],
                         t('Columns')
                     ));
-                }
 
-                $colCount++;
+                    $titleAdded = true;
+                }
 
                 yield $columnName => $columnMeta;
             }
         }
 
         // Finally, the other custom variable suggestions
-        if (! empty($varSuggestions)) {
-            $this->addHtml(HtmlElement::create(
-                'li',
-                ['class' => static::SUGGESTION_TITLE_CLASS],
-                t('Custom Variables')
-            ));
-
-            foreach ($varSuggestions as $relation => $label) {
-                yield $relation => $label;
+        $titleAdded = false;
+        foreach (
+            $this->getDb()->select($this->queryCustomvarConfig(Filter::like('flatname', $searchTerm))) as $customVar
+        ) {
+            $search = $name = $customVar->flatname;
+            if (preg_match('/\w+(?:\[(\d*)])+$/', $search, $matches)) {
+                $name = substr($search, 0, -(strlen($matches[1]) + 2));
+                if (isset($parsedArrayVars[$name])) {
+                    continue;
+                }
+
+                $parsedArrayVars[$name] = true;
+                $search = $name . '[*]';
+            }
+
+            foreach ($this->customVarSources as $relation => $label) {
+                if (isset($customVar->$relation)) {
+                    $varRelation = $relation . '.vars.' . $search;
+                    $varLabel = sprintf($label, $name);
+                    // Suggest exact custom variable matches first
+                    if (
+                        ! in_array($varRelation, $exactVarMatches)
+                        && $this->matchSuggestion($varRelation, $varLabel, $searchTerm)
```

Comment:
Please restore `filterColumnSuggestions`. Didn't test, but aren't columns not filtered right now?

Reply:
As discussed offline, the custom variables need not be filtered again. But `filterColumnSuggestions` should not be restored as we are filtering columns directly in `fetchColumnSuggestions`.